core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java

```diff
@@ -7043,6 +7043,14 @@ public void testAlterTableAddRowField()
                 .hasMessage("line 1:1: Adding fields with COMMENT is unsupported");
     }
 
+    @Test
+    public void testNNonBooleanFilterExpression()
```

Comment:
I would move to the above `testInvalidAggregationFilter` method. 
```suggestion
    public void testNonBooleanFilterExpression()
```

Reply:
Merged this case into `testInvalidAggregationFilter`.